### PR TITLE
CLN: move raising TypeError for `interpolate` with `object` dtype to Block

### PR DESCRIPTION
**Pull request by natmokval**
xref #57820

moved raising `TypeError` for `interpolate` with object dtype from ``NDFrame`` to ``Block``

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1388,12 +1388,13 @@ def interpolate(
...
             return [self.copy(deep=False)]
 
-        # TODO(3.0): this case will not be reachable once GH#53638 is enforced
         if self.dtype == _dtype_obj:
             # only deal with floats
             # bc we already checked that can_hold_na, we don't have int dtype here
             # test_interp_basic checks that we make a copy here
-            return [self.copy(deep=False)]
+            raise TypeError(
+                f"{type(self).__name__} cannot interpolate with object dtype."
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py:1396]: we dont need to have Block in the exception message. the comments in the line above can be removed.  can you add "# GH#53631"

Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:1396]: can do something like `name = {1: "Series", 2: "DataFrame"}[self.ndim]` to keep the exception message unchanged

Comment (natmokval) [reply pandas/core/internals/blocks.py]: thank you, I made changes, ci - green.

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Comment by jbrockmendel** [comment]
thanks @natmokval

**Pull request merged**